lib/thrift/binary/framed/client.ex

```diff
@@ -291,10 +302,43 @@ defmodule Thrift.Binary.Framed.Client do
     end
   end
 
+  @impl Connection
+  def handle_info({:tcp_closed, sock}, %{reconnect: true, sock: {:gen_tcp, sock}} = s) do
+    {:disconnect, :reconnect, s}
+  end
+
+  def handle_info({:ssl_closed, sock}, %{reconnect: true, sock: {:ssl, sock}} = s) do
+    {:disconnect, :reconnect, s}
+  end
+
+  def handle_info(_, s) do
```

Comment:
We should handle the `{:tcp_error, sock, _}`, `{:ssl_error, sock, _}`, `{:tcp, sock, _}, {;ssl, sock, _}` here but ignore messages from old sockets (or clean them up in disconnect/2).